src/hooks/core/chat/useChannelGetInitialMessagesHook.ts

```diff
@@ -34,27 +35,41 @@ const useChannelGetInitialMessagesHook = () => {
 
       if (!shouldVisibleItemGetInitialMessages) return;
 
-      queue.addPriorityJob({
-        operationLabel: DatabaseOperationLabel.ChannelList_GetInitialMessages,
-        id: item?.id,
-        priority: QueueJobPriority.HIGH,
-        task: async () => {
-          const chats = await ChatSchema.getAll(
-            localDb,
-            item?.id,
-            signedProfileId,
-            anonProfileId,
-            CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
-          );
+      if (Platform.OS === 'android') {
+        const chats = await ChatSchema.getAll(
+          localDb,
+          item?.id,
+          signedProfileId,
+          anonProfileId,
+          CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
+        );
 
-          return chats;
-        },
-        callback: (chats) => {
-          if (chats?.length > 0) {
-            initialMessagesHashMap.current[item?.id] = chats;
-          }
+        if (chats?.length > 0) {
+          initialMessagesHashMap.current[item?.id] = chats;
         }
-      });
+      } else {
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.ChannelList_GetInitialMessages,
+          id: item?.id,
+          priority: QueueJobPriority.HIGH,
+          task: async () => {
+            const chats = await ChatSchema.getAll(
+              localDb,
+              item?.id,
+              signedProfileId,
+              anonProfileId,
+              CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
+            );
+
+            return chats;
+          },
+          callback: (chats) => {
+            if (chats?.length > 0) {
+              initialMessagesHashMap.current[item?.id] = chats;
+            }
+          }
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of platform-specific logic here is a good approach to handle the differences in task processing between Android and other platforms. However, there's a redundancy in the code where the same operation `ChatSchema.getAll()` is performed in both conditions. This could be improved by extracting this operation into a separate function and reusing it.

```diff
-      if (Platform.OS === 'android') {
-        const chats = await ChatSchema.getAll(
-          localDb,
-          item?.id,
-          signedProfileId,
-          anonProfileId,
-          CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
-        );
- 
-        if (chats?.length > 0) {
-          initialMessagesHashMap.current[item?.id] = chats;
-         }
-      } else {
-        queue.addPriorityJob({
-          operationLabel: DatabaseOperationLabel.ChannelList_GetInitialMessages,
-          id: item?.id,
-          priority: QueueJobPriority.HIGH,
-          task: async () => {
-            const chats = await ChatSchema.getAll(
-              localDb,
-              item?.id,
-              signedProfileId,
-              anonProfileId,
-              CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
-            );
- 
-            return chats;
-          },
-          callback: (chats) => {
-            if (chats?.length > 0) {
-              initialMessagesHashMap.current[item?.id] = chats;
-            }
-          }
-        });
-      }

+      const getChats = async () => {
+        return await ChatSchema.getAll(
+          localDb,
+          item?.id,
+          signedProfileId,
+          anonProfileId,
+          CHANNEL_LIST_GET_INITIAL_MESSAGES_LIMIT
+        );
+      };
+
+      if (Platform.OS === 'android') {
+        const chats = await getChats();
+        if (chats?.length > 0) {
+          initialMessagesHashMap.current[item?.id] = chats;
+        }
+      } else {
+        queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.ChannelList_GetInitialMessages,
+          id: item?.id,
+          priority: QueueJobPriority.HIGH,
+          task: getChats,
+          callback: (chats) => {
+            if (chats?.length > 0) {
+              initialMessagesHashMap.current[item?.id] = chats;
+            }
+          }
+        });
+      }
```
This change reduces redundancy and improves maintainability by ensuring that any future changes to the `ChatSchema.getAll()` operation only need to be made in one place.


<!-- This is an auto-generated comment by OSS CodeRabbit -->